feat(dracut): disable xattr when creating cpios

jozzsi opened the pull request:
## Changes

Disable xattr when creating cpio, but do not change the default for squashfs/erofs.

Move setting the default for `DRACUT_CP` later in the code to account for the newly set defaults.

## Checklist
- [ ] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #1561

jozzsi committed:
feat(dracut): disable xattr when creating cpios

Disable xattr when creating cpio, but do not change
the default for squashfs/erofs.

Move setting the default for DRACUT_CP later in the code to
account for the newly set defaults.

Fixes: https://github.com/dracut-ng/dracut-ng/issues/1561

jozzsi commented:
> SELinux that depends on it

Is it ? From the referenced bug quoting @cgwalters 

>  am not aware of any use case for setting `security.selinux` at all in the initramfs; systemd only loads policy from the real root when transitioning.

Defaults are important and powerful. Distributions and users downstream started to flip this default already (like Fedora-bootc). What is the upside/benefit of having different Fedora based distributions running with different dracut defaults ?

From a quick look of it, it seems even cp does not preserve `xattr` by default. What makes `dracut` choosing a different default than `cp`? Is it really what most users want by default ?

cgwalters commented:
> SELinux is not what I'm worried about. I'm worried about everything else. [Since Fedora 15](https://fedoraproject.org/wiki/Features/RemoveSETUID), generally new stuff in Fedora uses file capabilities to get privileges as applications, and those are done as extended attributes on the filesystem too.

I think what needs to be emphasized heavily here is that the cpio format (and the Linux kernel's parser) *do not support xattrs* - they never have. There *are* however out of tree patches to do so, see https://github.com/dracut-ng/dracut-ng/issues/1561#issuecomment-3179438340 and https://lwn.net/Articles/747473/ etc.

dracut is just pointlessly trying to copy the xattrs to the staging directory, only to ignore them when generating the cpio.

jozzsi commented on dracut.conf.d/fedora/01-dist.conf:
```diff
@@ -16,3 +16,6 @@ hostonly="yes"
 reproducible="yes"
+
+# enable xattr
+export DRACUT_NO_XATTR=0
```
Thanks @cgwalters . Please take a second look, as the PR intent to do the opposite - change the default for all distributions except Fedora (as the Fedora maintainer seems to be concerned about making this change for Fedora). 

See the commit message

> Do not change the default for Fedora-based distributions and keep xattr enabled by default.

jozzsi replied:
Removed the Fedora specific changes based on latest [feedback](https://github.com/dracut-ng/dracut-ng/pull/1987#issuecomment-3712322724) from @Conan-Kudo .

jozzsi commented:
> so let's just disable it in code for cpio regardless of generator.

Ok, thanks @Conan-Kudo . Updated the PR and removed the Fedora specific changes.
